proto/ca_service.proto

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package proto;
+
+// Service definition of IstioCAService that can dynamically verify the CSR for
+// service identities for Istio services running on various platforms and
+// approve or disapprove the request after verifying the Node Agent credentials
+// provided as a part of the Certificte Signing Request.
+service IstioClusterCAService {
+
+  // Request to sign a Certificate Signing Request.
+  //
+  // The CSR is generated on the Node Agent along with the private key. The Node
+  // Agent also needs to provide its credentials so the CA can verify the Node
+  // Agent requesting the credentials.
+  rpc Sign(CertificateSignRequest) returns (CertificateSignResponse);
+}
+
+message CertificateSignRequest {
+  bytes csr = 1;
+  NodeAgentCredentials nodeAgentCredentials = 2;
+}
+
+message NodeAgentCredentials {
+  oneof node_credential {
+    GcpInstanceCredential gcp_instance_credential = 1;
+    OnPremNodeCredential on_prem_node_credential = 2;
+  }
+}
+
+message GcpInstanceCredential {
+  bytes signed_instance_metadata = 1;
+}
+
+message OnPremNodeCredential {
```

Comment:
The reason I added this is because I need to get the node identity from the cert. While we need to provide the identity on the channel, I will also need the identity on the cert to get the Node Identity so I can get the service identity for the node. I provided both the individual as well as cert chain since I was not fully sure if it is possible to provide the certs in both ways - i.e., individual cert and cert chain. Another option is for the Node Agent to provide the identity but I am not too sure if this is something the Node Agent should do. It just provides the cert / cert-chain to the CA and the CA will sign the CSR with the service identity.